pkg/logs/client/tcp/destination.go

```diff
@@ -64,13 +72,16 @@ func (d *Destination) Send(payload []byte) error {
 		return client.NewFramingError(err)
```

Comment:
In that case, we don't put back the connection into the pool. It is maybe better to first construct the frame and then get or create a connection.

Reply:
good catch